app/models/king.rb

```diff
@@ -1,7 +1,68 @@
+# rubocop:disable Metrics/AbcSize
 class King < Piece
   def move_legal?(to_row, to_col)
     row_diff = (row - to_row).abs
     col_diff = (col - to_col).abs
     row_diff <= 1 && col_diff <= 1
+    # add castling moves to the 'whitelist', then verify the validity of the castle
+    #   further down in the logic
+  end
+
+  # override to enable castling
+  def finalize_move!(to_row, to_col)
+    return false if !get_attackers(to_row, to_col).empty?
+    move_to!(to_row, to_col)
+    # for castling something like: move_the_rook if castling?
+  end
+
+  def in_check?
+    !get_attackers(row, col).empty?
```

Comment:
I chose not to use `any?` here because it's not strictly equivalent to `!...empty?`. `any?` roughly translates to "is there at least one non-false and non-nil element in the array?" This means `any?` is generally slower (though only this is only noticeable for larger arrays), ant it will return false if there is any array of only nil and/or false elements:
```
[nil, nil, false].any? # false
[nil, nil, false].empty? # false
```

In general, I try not to use `any?` if I'm not using it with a block.  In my view, it's worth the slight decrease in readability for this case.